tooling/nargo_toml/src/flock.rs

```diff
@@ -0,0 +1,33 @@
+use fs2::FileExt;
+use std::{
+    fs::{File, OpenOptions},
+    path::Path,
+};
+
+// TODO: move this to some utils crate.
+
+pub(crate) struct FileLock {
+    file: File,
+}
+
+impl FileLock {
+    pub(crate) fn new(file_path: &Path, lock_name: &str) -> std::io::Result<Self> {
+        std::fs::create_dir_all(file_path.parent().expect("can't create lock on filesystem root"))?;
+        let file = OpenOptions::new().create(true).truncate(true).write(true).open(file_path)?;
+        if file.try_lock_exclusive().is_err() {
+            eprintln!("Waiting for lock on {lock_name}...");
```

Comment:
I think printing the full path here would be fine, or as I mentioned just the name of the lock file.

Reply:
I think it would be more user-friendly to have a custom name used here for when we want a directory lock.

Reply:
Okay, I suppose at least very cryptic file names which only work in the full path context can be avoided this way.